migrate/list-convert.go

```diff
@@ -0,0 +1,181 @@
+		NewDrive.ObjectMeta.DeletionTimestamp = OldDrive.ObjectMeta.DeletionTimestamp
+		NewDrive.ObjectMeta.DeletionGracePeriodSeconds = OldDrive.ObjectMeta.DeletionGracePeriodSeconds
+		NewDrive.ObjectMeta.Labels = OldDrive.ObjectMeta.Labels
+		NewDrive.ObjectMeta.Annotations = OldDrive.ObjectMeta.Annotations
+		NewDrive.ObjectMeta.OwnerReferences = OldDrive.ObjectMeta.OwnerReferences
+		NewDrive.ObjectMeta.Finalizers = OldDrive.ObjectMeta.Finalizers
+		NewDrive.ObjectMeta.ZZZ_DeprecatedClusterName = OldDrive.ObjectMeta.ZZZ_DeprecatedClusterName
+		NewDrive.ObjectMeta.ManagedFields = OldDrive.ObjectMeta.ManagedFields
+
+		// Status fields
+		NewDrive.Status.Path = OldDrive.Status.Path
+		if OldDrive.Status.DriveStatus == "InUse" || OldDrive.Status.DriveStatus == "Ready" {
+			NewDrive.Status.Status = "OK"
+		}
+		NewDrive.Status.TotalCapacity = OldDrive.Status.TotalCapacity
+		NewDrive.Status.AllocatedCapacity = OldDrive.Status.AllocatedCapacity
+		NewDrive.Status.FreeCapacity = OldDrive.Status.FreeCapacity
+		NewDrive.Status.FSUUID = OldDrive.Status.FilesystemUUID
+		NewDrive.Status.NodeName = OldDrive.Status.NodeName
+		NewDrive.Status.Topology = OldDrive.Status.Topology
+		NewDrive.Status.ModelNumber = OldDrive.Status.ModelNumber
+		NewDrive.Status.Vendor = OldDrive.Status.Vendor
+		AccessTier := OldDrive.Status.AccessTier
+		if AccessTier == "Warm" {
+			NewDrive.Status.AccessTier = "Warm"
+		}
+		if AccessTier == "Hot" {
+			NewDrive.Status.AccessTier = "Hot"
+		}
+		if AccessTier == "Cold" {
+			NewDrive.Status.AccessTier = "Cold"
+		}
+		if AccessTier == "Unknown" {
+			NewDrive.Status.AccessTier = "Unknown"
+		}
+
+		// Append the new drive to the list
+		newDrives = append(newDrives, NewDrive)
+	}
+
+	// Return the new list of drives
+	return newDrives, nil
+}
+
+func main() {
+
+	// Get the kubernetes configuration file
+	kubeConfig, err := getKubeConfig()
+	if err != nil {
+		fmt.Printf("%s: Could not connect to kubernetes. %s=%s\n", "Error", "KUBECONFIG", kubeConfig)
+		os.Exit(1)
+	}
+
+	// Get the latest interface available
+	latestDirectCSIDriveInterface, err := directcsiclient.DirectCSIDriveInterfaceForConfig(kubeConfig)
+	if err != nil {
+		fmt.Printf("%s: could not initialize drive adapter client: err=%v\n", "Error", err)
+		os.Exit(1)
+	}
+
+	// Then we list the drives
+	// This is getting all the info from all the drives
+	ctx, _ := context.WithCancel(context.Background())
+	driveList, err := latestDirectCSIDriveInterface.List(ctx, metav1.ListOptions{})
+	if err != nil {
+		fmt.Println("error")
+	}
+
+	// To filter the drives by status and version:
+	filteredDrives := []directcsi.DirectCSIDrive{}
+	for _, drive := range driveList.Items {
+		// Only for Ready and InUse status are valid here for this list of migration:
+		if drive.Status.DriveStatus == "Ready" || drive.Status.DriveStatus == "InUse" {
+			// Only for version v1beta5 is applicable the migration
+			if drive.TypeMeta.APIVersion == "direct.csi.min.io/v1beta5" {
```

Comment:
we need to run discovery to find out the storage version of the object - https://github.com/minio/directpv/blob/master/pkg/client/interface.go#L79

because, `latestDirectCSIDriveInterface.List` uses dynamic clients, which will dynamically convert all the older objects to the latest ones. So the older objects will always have the `v1beta5` as its apiVersion as it comes through dynamic conversion.

Reply:
Ok, I am now using `GetGroupKindVersions()` to get the values. Hope this is what is needed.